package/libpiksi/libpiksi/src/logging.c

```diff
@@ -73,39 +83,29 @@ void sbp_log(int priority, const char *msg_text, ...)
   va_end(ap);
 }
 
-void sbp_vlog(int priority, const char *msg_text, va_list ap)
+void sbp_vlog(int priority, const char *msg, va_list ap)
 {
-  const char *log_args[NUM_LOG_LEVELS] =
-    {"emerg", "alert", "crit", "error", "warn", "notice", "info", "debug"};
+  msg_log_t *log;
+  char buf[SBP_FRAMING_MAX_PAYLOAD_SIZE];
 
-  if (priority < 0 || priority >= NUM_LOG_LEVELS) {
-    priority = LOG_INFO;
+  if (!sbp_tx) {
+    piksi_log(LOG_ERR, "attempt to send SBP prior to initialization.");
+    return;
   }
 
-  char cmd_buf[256];
-  snprintf(cmd_buf, sizeof(cmd_buf), "sbp_log --%s", log_args[priority]);
-  FILE *output = popen(cmd_buf, "w");
-
-  if (output == 0) {
-    piksi_log(LOG_ERR, "couldn't call sbp_log.");
+  if (priority < 0 || priority > UINT8_MAX) {
+    piksi_log(LOG_ERR, "invalid SBP log level.");
     return;
   }
 
-  char formatted_msg[2048];
-
-  vsnprintf(formatted_msg, sizeof(formatted_msg), msg_text, ap);
+  log = (msg_log_t *)buf;
+  log->level = (uint8_t)priority;
 
-  char msg_buf[2048];
-  snprintf(msg_buf, sizeof(msg_buf), "%s: %s", log_ident, formatted_msg);
+  int n = vsnprintf(log->text, SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(msg_log_t), msg, ap);
 
-  fputs(msg_buf, output);
+  if (n < 0) return;
```

Comment:
Do we want an explicit error logged in this case?

Reply:
I think we'll be OK without it, at the point where we have code that needs to react with sbp_log fails we can add a return code.